### ISSUE TITLE
Can't build app with OpenSSL 3.0.2 (static) on Android x86

### ISSUE THREAD

**Issue by AenBleidd** [openssl/openssl#18059]
```
configure:17890: checking whether libcurl is usable
configure:17923: i686-linux-android16-clang -o conftest --sysroot=/home/runner/work/boinc/boinc/3rdParty/android/android-ndk-r21d/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DANDROID -DDECLARE_TIMEZONE -Wall -I/home/runner/work/boinc/boinc/3rdParty/buildCache/android/android-tc/x86/i686-linux-android/include -O3 -fomit-frame-pointer -fPIE -D__ANDROID_API__=16 -Wall  -I/home/runner/work/boinc/boinc/3rdParty/android/vcpkg/installed/x86-android/include -I/home/runner/work/boinc/boinc/3rdParty/android/vcpkg/installed/x86-android/lib/pkgconfig/../../include -L/home/runner/work/boinc/boinc/3rdParty/android/vcpkg/installed/x86-android/lib -llog -fPIE -pie -latomic -static-libstdc++ -L/home/runner/work/boinc/boinc/3rdParty/android/vcpkg/installed/x86-android/lib conftest.c  -L/home/runner/work/boinc/boinc/3rdParty/android/vcpkg/installed/x86-android/lib/pkgconfig/../../lib -lcurl -lssl -lcrypto -ldl -pthread -lz >&5
clang: warning: argument unused during compilation: '-static-libstdc++' [-Wunused-command-line-argument]
crypto/threads_pthread.c:219: error: undefined reference to '__atomic_is_lock_free'
crypto/threads_pthread.c:220: error: undefined reference to '__atomic_fetch_or_8'
crypto/threads_pthread.c:244: error: undefined reference to '__atomic_is_lock_free'
crypto/threads_pthread.c:245: error: undefined reference to '__atomic_load'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by AenBleidd** [comment]
Just for reference, `-latomic` is provided but still doesn't work for some reason

**Comment by AenBleidd** [comment]
@VitaliLiashchuk, I updated the NDK, and the issue has gone: https://github.com/BOINC/boinc/pull/4751

**Comment by VitaliLiashchuk** [comment]
You save my day! Thanks for the tip! 

I updated ndk 21.4.7075529 to ndk 23.2.8568313 it solved the problem.

**Comment by nhorman** [comment]
Marking as inactive, will be closed at the end of 3.4 dev  without further input.
